thorlcr/thorutil/thormisc.hpp

```diff
@@ -603,13 +603,15 @@ inline void readUnderlyingType(MemoryBuffer &mb, T &v)
 constexpr unsigned thorDetailedLogLevel = 200;
 constexpr LogMsgCategory MCthorDetailedDebugInfo(MCdebugInfo(thorDetailedLogLevel));
 
-class CThorContextLogger : public CDefaultContextLogger<IContextLogger>
+class CThorContextLogger : public CDefaultContextLogger<IContextLogger>, public CInterface
 {
     unsigned traceLevel = 1;
     LogTrace logTrace;
     mutable CRuntimeStatisticCollection stats;
 
 public:
+    IMPLEMENT_IINTERFACE;
```

Comment:
If I remove this, I get a compilation error:

/home/shamser/HPCC-Platform/thorlcr/graph/./../thorutil/thormisc.hpp:606:7: note:   because the following virtual functions are pure within ‘CThorContextLogger’:
  606 | class CThorContextLogger : public CDefaultContextLogger<IContextLogger>
      |       ^~~~~~~~~~~~~~~~~~
In file included from /home/shamser/HPCC-Platform/thorlcr/graph/./../../system/jlib/jiface.hpp:27,
                 from /home/shamser/HPCC-Platform/thorlcr/graph/./../../system/jlib/jlib.hpp:34,
                 from /home/shamser/HPCC-Platform/thorlcr/graph/thgraph.hpp:43,
                 from /home/shamser/HPCC-Platform/thorlcr/graph/thgraph.cpp:18:
/home/shamser/HPCC-Platform/thorlcr/graph/./../../system/jlib/jscm.hpp:43:18: note:     ‘virtual void IInterface::Link() const’
   43 |     virtual void Link() const = 0;
      |                  ^~~~
/home/shamser/HPCC-Platform/thorlcr/graph/./../../system/jlib/jscm.hpp:44:18: note:     ‘virtual bool IInterface::Release() const’
   44 |     virtual bool Release() const = 0;

Reply:
that's because the base class CDefaultContextLogger<IContextLogger> is using double inheritance and does not itself implement the pure virtual IInterface::Release/Link.

In its current form, it (the base class) should use IMPLEMENT_INTERFACE,
but it would be better to avoid the double inheritance there, by making it:
```
class CDefaultContextLogger : public CInterfaceOf<INTERFACE>
```